scripts/prune-sstate-cache/prune_cache.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python3
+
+import datetime
+import os
+import sys
+import argparse
+
+def num_bytes(N):
+    suffix = ["B", "KiB", "MiB", "GiB", "TiB", "UWOT_M8?!?"]
+    tmp = float(N)
+    for s in suffix:
+        if tmp / 1024 < 1.0:
+            return tmp, s
+        tmp /= 1024
+    return tmp, "UWOT?"
+
+def main(args):
+    root = os.path.abspath(args.dir)
+    try:
+        days = int(args.time)
+    except ValueError:
+        print("Usage: python %s <dir_base> <num_days_old>" % argv[0])
+        return
+    
+    timelimit = datetime.datetime.now() - datetime.timedelta(days=days)
+
+    objects = [] #(timestamp, path, file_size)
+    for path, dirnames, filenames  in os.walk(root, topdown=True):
+        for filename in filenames:
+            file_path = os.path.join(path, filename)
+            if os.path.islink(file_path):
+                continue
+            obj = (
+                datetime.datetime.fromtimestamp(os.path.getmtime(file_path)),
+                file_path,
+                os.path.getsize(file_path)
+            )
+            objects.append(obj)
+        for dirname in dirnames:
+            dir_path = os.path.join(path, dirname)
+            if os.path.islink(dir_path):
+                continue
+            obj = None
+            if len(os.listdir(dir_path)) == 0:
+                obj = (
+                    datetime.datetime.fromtimestamp(os.path.getmtime(dir_path)),
+                    dir_path,
+                    os.path.getsize(dir_path)
+                )
+            else:
+                continue
```

Comment:
I can see this loop does not remove broken links and empty directories...

Reply:
I don't think I dare to touch this script... Could you be so kind to launch a PR after I merge this one?